packages/grid/_modules_/grid/models/gridOptions.tsx

```diff
@@ -277,6 +277,11 @@ export interface GridOptions {
    * @param param With all properties from [[GridColParams]].
    */
   onColumnHeaderLeave?: (param: GridColParams, event: React.MouseEvent) => void;
+  /**
+   * Callback fired when a column is reordered.
+   * @param param With all properties from [[GridColParams]].
```

Comment:
?

```suggestion
   * @param param With all properties from [[GridColParams]].
   * @param event [[React.MouseEvent]].
```

Reply:
My bad. There are other methods without the description for the 'event' argument as well in this file.

Reply:
`OnColumnReorder` could be fired differently, the signature should be 

`onColumnReorder(params: GridColumnHeaderParams, event?: React.SyntheticEvent);`

Reply:
At this moment the `event` argument is always available because `onColumnReorder` is fired by the `dragend` event. It could be optional if an `api.setColumnIndex` method gets exposed, but that raises me the doubt if this new method is necessary because `api.moveColumn` is already available.

Reply:
yes so `moveColumn` is a bit misleading, I suggest we rename it to `setColumnIndex` which should do the same under the hood. `SetColumnIndex` should publish `ColumnReorder` and `GRID_COLUMNS_UPDATED`. Params for the later can be updated too